hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -362,18 +362,10 @@ private NavigableSet<ContainerID> getContainersForOwner(
       Pipeline pipeline, String owner) throws IOException {
     NavigableSet<ContainerID> containerIDs =
         pipelineManager.getContainersInPipeline(pipeline.getId());
-    Iterator<ContainerID> containerIDIterator = containerIDs.iterator();
-    while (containerIDIterator.hasNext()) {
-      ContainerID cid = containerIDIterator.next();
-      try {
-        if (!getContainer(cid).getOwner().equals(owner)) {
-          containerIDIterator.remove();
-        }
-      } catch (ContainerNotFoundException e) {
-        LOG.error("Could not find container info for container {}", cid, e);
-        containerIDIterator.remove();
-      }
-    }
+    containerIDs = new TreeSet<>(containerIDs);
```

Comment:
Why are you creating a `new TreeSet<>(containerIDs)` here? The call to `pipelineManager.getContainersInPipeline()` already returns the new TreeSet.

Reply:
> Why are you creating a `new TreeSet<>(containerIDs)` here? The call to `pipelineManager.getContainersInPipeline()` already returns the new TreeSet.

Sorry, it really should be removed, I didn't notice it